File: demo/CircleTimer/ViewController.m
//
//  ViewController.m
//  CircleTimer
//
// Created by Kirill Serebriakov on 9/25/14.
// Copyright (c) 2014 Appus Studio LLC. All rights reserved.
//

#import "ViewController.h"
#import "CircleTimer.h"

@interface ViewController () <CircleTimerDelegate>

@property (weak, nonatomic) IBOutlet CircleTimer *timeCircle;


@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.timeCircle.delegate = self;
    self.timeCircle.active = YES;
    [self updateCircleTimer];

}

- (void)updateCircleTimer {
    
    self.timeCircle.totalTime = 60;
    self.timeCircle.elapsedTime = 20;
}

#pragma mark - Action


- (IBAction)playStopClicked:(UIButton *)sender {
    
    sender.selected = !sender.selected;
    
    if(sender.selected){
        if(self.timeCircle.didStart){
            [self.timeCircle resume];
        }else{
            [self.timeCircle start];
        }
    }else{
        [self.timeCircle stop];
    }
}




@end

File: demo/CircleTimer/ViewController.swift
//
//  ViewController.swift
//  CircleTimer
//
//  Created by Sergey Sokoltsov on 11/30/16.
//  Copyright © 2016 Sergey Sokoltsov. All rights reserved.
//

import UIKit
import AppusCircleTimer

class ViewController: UIViewController, AppusCircleTimerDelegate {
    @IBOutlet weak var circleTimer : AppusCircleTimer?
    @IBOutlet weak var startButton : UIButton?
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        circleTimer?.delegate = self
        circleTimer?.isBackwards = true
        circleTimer?.isActive = true
        circleTimer?.totalTime = 60
        circleTimer?.elapsedTime = 58
        // Do any additional setup after loading the view, typically from a nib.
    }

    func circleCounterTimeDidExpire(circleTimer: AppusCircleTimer) {
        startButton?.isSelected = false
    }


    @IBAction func startStopClicked(sender:UIButton) {
        sender.isSelected = !sender.isSelected;
        if(sender.isSelected){
            if(circleTimer?.didStart)!{
                circleTimer?.resume()
            } else {
                circleTimer?.start()
            }
        }else{
            circleTimer?.stop()
        }
    }
    
    @IBAction func resetClicked(_ sender: Any) {
        circleTimer?.reset()
        circleTimer?.elapsedTime = 58
        startButton?.isSelected = false
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

